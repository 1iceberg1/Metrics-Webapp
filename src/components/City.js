import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import citiesData from '../redux/citiesData';
import { fetchCities } from '../redux/city/citiesSlice';
import cloudy from '../assets/cloudy.png';

const City = () => {
  const [keyword, setKeyword] = useState('');

  const handleChange = (e) => {
    setKeyword(e.target.value);
  };

  const cities = useSelector((state) => state.cities);
  const dispatch = useDispatch();

  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(fetchCities(keyword));
    setKeyword('');
  };

  return (
    <>
      <form onSubmit={handleSubmit}>
        <input
          type="text"
          placeholder="Search City"
          required
          value={keyword}
          onChange={handleChange}
        />
        <button type="submit">Search</button>
      </form>
      <section>
        {!cities
        && citiesData.map((city) => (
          <Link
            key={city.country}
            to={{
              pathname: `/detail/${city.name}/${city.lat}/${city.lon}`,
            }}
          >
            <div className="city">
              <div className="city-img"><img src={cloudy} alt="country map" /></div>
              <div className="city-info">
                <h2>{city.name}</h2>
                <p className="coordinate">
                  <span>Lat: </span>
                  <span>{city.lat}</span>
                </p>
                <p className="coordinate">
                  <span>Long: </span>
                  <span>{city.lon}</span>
                </p>
              </div>
            </div>
          </Link>
        ))}
        {cities && cities.coord && (
          <Link
            key={cities.id}
            to={{
              pathname: `/detail/${cities.name}/${cities.coord.lat}/${cities.coord.lon}`,
            }}
          >
            <div className="city">
              <img src={cloudy} alt="weather icon" />
              <div className="city-info">
                <h2>{cities.name}</h2>
                <p className="coordinate">
                  <span>Lat: </span>
                  <span>{cities.coord.lat}</span>
                </p>
                <p className="coordinate">
                  <span>Long: </span>
                  <span>{cities.coord.lon}</span>
                </p>
              </div>
            </div>
          </Link>
        )}
        {cities && cities.cod && (
          <>
            <h1>{cities.message}</h1>
          </>
        )}
      </section>
    </>
  );
};

export default City;
