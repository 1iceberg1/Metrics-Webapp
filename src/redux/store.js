import { configureStore } from '@reduxjs/toolkit';
import detailsSlice from './detail/detailsSlice';
import citiesSlice from './city/citiesSlice';

const store = configureStore({
  reducer: {
    details: detailsSlice,
    cities: citiesSlice,
  },
});

export default store;
