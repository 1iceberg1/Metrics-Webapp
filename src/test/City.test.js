import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import City from '../components/City';

describe('App component', () => {
  test('renders', () => {
    const tree = render(
      <Provider store={store}>
        <Router>
          <City />
        </Router>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
