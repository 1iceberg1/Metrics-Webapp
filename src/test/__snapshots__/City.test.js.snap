// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section>
        <a
          href="/detail/London/44.34/10.99"
        >
          <div
            class="city"
          >
            <img
              alt="country map"
              src="https://raw.githubusercontent.com/djaiss/mapsicon/master/europe/gb/512.png"
            />
            <div
              class="city-info"
            >
              <h2>
                London
              </h2>
              <p
                class="latitude"
              >
                <span>
                  Lat: 
                </span>
                <span>
                  44.34
                </span>
              </p>
              <p
                class="longitude"
              >
                <span>
                  Long: 
                </span>
                <span>
                  10.99
                </span>
              </p>
            </div>
          </div>
        </a>
        <a
          href="/detail/Paris/48.8589/2.32"
        >
          <div
            class="city"
          >
            <img
              alt="country map"
              src="https://raw.githubusercontent.com/djaiss/mapsicon/master/europe/fr/512.png"
            />
            <div
              class="city-info"
            >
              <h2>
                Paris
              </h2>
              <p
                class="latitude"
              >
                <span>
                  Lat: 
                </span>
                <span>
                  48.8589
                </span>
              </p>
              <p
                class="longitude"
              >
                <span>
                  Long: 
                </span>
                <span>
                  2.32
                </span>
              </p>
            </div>
          </div>
        </a>
        <a
          href="/detail/Rome/41.8933/12.4829"
        >
          <div
            class="city"
          >
            <img
              alt="country map"
              src="https://raw.githubusercontent.com/djaiss/mapsicon/master/europe/gr/512.png"
            />
            <div
              class="city-info"
            >
              <h2>
                Rome
              </h2>
              <p
                class="latitude"
              >
                <span>
                  Lat: 
                </span>
                <span>
                  41.8933
                </span>
              </p>
              <p
                class="longitude"
              >
                <span>
                  Long: 
                </span>
                <span>
                  12.4829
                </span>
              </p>
            </div>
          </div>
        </a>
        <a
          href="/detail/Spain/40.4167/-3.7036"
        >
          <div
            class="city"
          >
            <img
              alt="country map"
              src="https://raw.githubusercontent.com/djaiss/mapsicon/master/europe/es/512.png"
            />
            <div
              class="city-info"
            >
              <h2>
                Spain
              </h2>
              <p
                class="latitude"
              >
                <span>
                  Lat: 
                </span>
                <span>
                  40.4167
                </span>
              </p>
              <p
                class="longitude"
              >
                <span>
                  Long: 
                </span>
                <span>
                  -3.7036
                </span>
              </p>
            </div>
          </div>
        </a>
        <a
          href="/detail/Berlin/52.517/13.3889"
        >
          <div
            class="city"
          >
            <img
              alt="country map"
              src="https://raw.githubusercontent.com/djaiss/mapsicon/master/europe/de/512.png"
            />
            <div
              class="city-info"
            >
              <h2>
                Berlin
              </h2>
              <p
                class="latitude"
              >
                <span>
                  Lat: 
                </span>
                <span>
                  52.517
                </span>
              </p>
              <p
                class="longitude"
              >
                <span>
                  Long: 
                </span>
                <span>
                  13.3889
                </span>
              </p>
            </div>
          </div>
        </a>
        <a
          href="/detail/Tokyo/35.6767/139.764"
        >
          <div
            class="city"
          >
            <img
              alt="country map"
              src="https://raw.githubusercontent.com/djaiss/mapsicon/master/asia/jp/512.png"
            />
            <div
              class="city-info"
            >
              <h2>
                Tokyo
              </h2>
              <p
                class="latitude"
              >
                <span>
                  Lat: 
                </span>
                <span>
                  35.6767
                </span>
              </p>
              <p
                class="longitude"
              >
                <span>
                  Long: 
                </span>
                <span>
                  139.764
                </span>
              </p>
            </div>
          </div>
        </a>
        <a
          href="/detail/Beijing/39.9062/116.3913"
        >
          <div
            class="city"
          >
            <img
              alt="country map"
              src="https://raw.githubusercontent.com/djaiss/mapsicon/master/asia/cn/512.png"
            />
            <div
              class="city-info"
            >
              <h2>
                Beijing
              </h2>
              <p
                class="latitude"
              >
                <span>
                  Lat: 
                </span>
                <span>
                  39.9062
                </span>
              </p>
              <p
                class="longitude"
              >
                <span>
                  Long: 
                </span>
                <span>
                  116.3913
                </span>
              </p>
            </div>
          </div>
        </a>
        <a
          href="/detail/Singapore/1.2899/103.8519"
        >
          <div
            class="city"
          >
            <img
              alt="country map"
              src="https://raw.githubusercontent.com/djaiss/mapsicon/master/asia/sg/512.png"
            />
            <div
              class="city-info"
            >
              <h2>
                Singapore
              </h2>
              <p
                class="latitude"
              >
                <span>
                  Lat: 
                </span>
                <span>
                  1.2899
                </span>
              </p>
              <p
                class="longitude"
              >
                <span>
                  Long: 
                </span>
                <span>
                  103.8519
                </span>
              </p>
            </div>
          </div>
        </a>
        <a
          href="/detail/Mumbai/19.0785/72.8782"
        >
          <div
            class="city"
          >
            <img
              alt="country map"
              src="https://raw.githubusercontent.com/djaiss/mapsicon/master/asia/in/512.png"
            />
            <div
              class="city-info"
            >
              <h2>
                Mumbai
              </h2>
              <p
                class="latitude"
              >
                <span>
                  Lat: 
                </span>
                <span>
                  19.0785
                </span>
              </p>
              <p
                class="longitude"
              >
                <span>
                  Long: 
                </span>
                <span>
                  72.8782
                </span>
              </p>
            </div>
          </div>
        </a>
        <a
          href="/detail/New York/39.6853/-93.9269"
        >
          <div
            class="city"
          >
            <img
              alt="country map"
              src="https://raw.githubusercontent.com/djaiss/mapsicon/master/us/us/512.png"
            />
            <div
              class="city-info"
            >
              <h2>
                New York
              </h2>
              <p
                class="latitude"
              >
                <span>
                  Lat: 
                </span>
                <span>
                  39.6853
                </span>
              </p>
              <p
                class="longitude"
              >
                <span>
                  Long: 
                </span>
                <span>
                  -93.9269
                </span>
              </p>
            </div>
          </div>
        </a>
      </section>
    </div>
  </body>,
  "container": <div>
    <section>
      <a
        href="/detail/London/44.34/10.99"
      >
        <div
          class="city"
        >
          <img
            alt="country map"
            src="https://raw.githubusercontent.com/djaiss/mapsicon/master/europe/gb/512.png"
          />
          <div
            class="city-info"
          >
            <h2>
              London
            </h2>
            <p
              class="latitude"
            >
              <span>
                Lat: 
              </span>
              <span>
                44.34
              </span>
            </p>
            <p
              class="longitude"
            >
              <span>
                Long: 
              </span>
              <span>
                10.99
              </span>
            </p>
          </div>
        </div>
      </a>
      <a
        href="/detail/Paris/48.8589/2.32"
      >
        <div
          class="city"
        >
          <img
            alt="country map"
            src="https://raw.githubusercontent.com/djaiss/mapsicon/master/europe/fr/512.png"
          />
          <div
            class="city-info"
          >
            <h2>
              Paris
            </h2>
            <p
              class="latitude"
            >
              <span>
                Lat: 
              </span>
              <span>
                48.8589
              </span>
            </p>
            <p
              class="longitude"
            >
              <span>
                Long: 
              </span>
              <span>
                2.32
              </span>
            </p>
          </div>
        </div>
      </a>
      <a
        href="/detail/Rome/41.8933/12.4829"
      >
        <div
          class="city"
        >
          <img
            alt="country map"
            src="https://raw.githubusercontent.com/djaiss/mapsicon/master/europe/gr/512.png"
          />
          <div
            class="city-info"
          >
            <h2>
              Rome
            </h2>
            <p
              class="latitude"
            >
              <span>
                Lat: 
              </span>
              <span>
                41.8933
              </span>
            </p>
            <p
              class="longitude"
            >
              <span>
                Long: 
              </span>
              <span>
                12.4829
              </span>
            </p>
          </div>
        </div>
      </a>
      <a
        href="/detail/Spain/40.4167/-3.7036"
      >
        <div
          class="city"
        >
          <img
            alt="country map"
            src="https://raw.githubusercontent.com/djaiss/mapsicon/master/europe/es/512.png"
          />
          <div
            class="city-info"
          >
            <h2>
              Spain
            </h2>
            <p
              class="latitude"
            >
              <span>
                Lat: 
              </span>
              <span>
                40.4167
              </span>
            </p>
            <p
              class="longitude"
            >
              <span>
                Long: 
              </span>
              <span>
                -3.7036
              </span>
            </p>
          </div>
        </div>
      </a>
      <a
        href="/detail/Berlin/52.517/13.3889"
      >
        <div
          class="city"
        >
          <img
            alt="country map"
            src="https://raw.githubusercontent.com/djaiss/mapsicon/master/europe/de/512.png"
          />
          <div
            class="city-info"
          >
            <h2>
              Berlin
            </h2>
            <p
              class="latitude"
            >
              <span>
                Lat: 
              </span>
              <span>
                52.517
              </span>
            </p>
            <p
              class="longitude"
            >
              <span>
                Long: 
              </span>
              <span>
                13.3889
              </span>
            </p>
          </div>
        </div>
      </a>
      <a
        href="/detail/Tokyo/35.6767/139.764"
      >
        <div
          class="city"
        >
          <img
            alt="country map"
            src="https://raw.githubusercontent.com/djaiss/mapsicon/master/asia/jp/512.png"
          />
          <div
            class="city-info"
          >
            <h2>
              Tokyo
            </h2>
            <p
              class="latitude"
            >
              <span>
                Lat: 
              </span>
              <span>
                35.6767
              </span>
            </p>
            <p
              class="longitude"
            >
              <span>
                Long: 
              </span>
              <span>
                139.764
              </span>
            </p>
          </div>
        </div>
      </a>
      <a
        href="/detail/Beijing/39.9062/116.3913"
      >
        <div
          class="city"
        >
          <img
            alt="country map"
            src="https://raw.githubusercontent.com/djaiss/mapsicon/master/asia/cn/512.png"
          />
          <div
            class="city-info"
          >
            <h2>
              Beijing
            </h2>
            <p
              class="latitude"
            >
              <span>
                Lat: 
              </span>
              <span>
                39.9062
              </span>
            </p>
            <p
              class="longitude"
            >
              <span>
                Long: 
              </span>
              <span>
                116.3913
              </span>
            </p>
          </div>
        </div>
      </a>
      <a
        href="/detail/Singapore/1.2899/103.8519"
      >
        <div
          class="city"
        >
          <img
            alt="country map"
            src="https://raw.githubusercontent.com/djaiss/mapsicon/master/asia/sg/512.png"
          />
          <div
            class="city-info"
          >
            <h2>
              Singapore
            </h2>
            <p
              class="latitude"
            >
              <span>
                Lat: 
              </span>
              <span>
                1.2899
              </span>
            </p>
            <p
              class="longitude"
            >
              <span>
                Long: 
              </span>
              <span>
                103.8519
              </span>
            </p>
          </div>
        </div>
      </a>
      <a
        href="/detail/Mumbai/19.0785/72.8782"
      >
        <div
          class="city"
        >
          <img
            alt="country map"
            src="https://raw.githubusercontent.com/djaiss/mapsicon/master/asia/in/512.png"
          />
          <div
            class="city-info"
          >
            <h2>
              Mumbai
            </h2>
            <p
              class="latitude"
            >
              <span>
                Lat: 
              </span>
              <span>
                19.0785
              </span>
            </p>
            <p
              class="longitude"
            >
              <span>
                Long: 
              </span>
              <span>
                72.8782
              </span>
            </p>
          </div>
        </div>
      </a>
      <a
        href="/detail/New York/39.6853/-93.9269"
      >
        <div
          class="city"
        >
          <img
            alt="country map"
            src="https://raw.githubusercontent.com/djaiss/mapsicon/master/us/us/512.png"
          />
          <div
            class="city-info"
          >
            <h2>
              New York
            </h2>
            <p
              class="latitude"
            >
              <span>
                Lat: 
              </span>
              <span>
                39.6853
              </span>
            </p>
            <p
              class="longitude"
            >
              <span>
                Long: 
              </span>
              <span>
                -93.9269
              </span>
            </p>
          </div>
        </div>
      </a>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
